### PR TITLE
Make sounds for admins when prayers occur.

### PR DESCRIPTION
**Pull request by IProduceWidgets**
## About the PR

Makes prayers and phones play sound for admins.

## Why / Balance

So theyre harder to miss.

## Technical details

Made the sound file ~~and volume~~ configurable per prototype. Also added a configurable cooldown for them.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

no cl no fun

### CODE REVIEW

**Comment by IProduceWidgets** [comment]
addresses [#18092]

**Comment by IProduceWidgets** [comment]
Test fail is random closet fill arbitrage, and unrelated.

File: Content.Server/Prayer/PrayerSystem.cs [hunk 1]
```diff
@@ -104,5 +111,13 @@ public void Pray(ICommonSession sender, PrayableComponent comp, string message)
 
         _chatManager.SendAdminAnnouncement($"{Loc.GetString(comp.NotificationPrefix)} <{sender.Name}>: {message}");
         _adminLogger.Add(LogType.AdminMessage, LogImpact.Low, $"{ToPrettyString(sender.AttachedEntity.Value):player} sent prayer ({Loc.GetString(comp.NotificationPrefix)}): {message}");
+		
+        /// Admin sound
+        if (_gameTiming.CurTime > comp.NextRing) {
+            _audioSystem.PlayGlobal(comp.SoundForGod,
+                Filter.Empty().AddPlayers(_adminManager.ActiveAdmins), false, AudioParams.Default.WithVolume(comp.VolumeForGod));
+            comp.NextRing = _gameTiming.CurTime + comp.RingPeriod;
```
Comment (metalgearsloth) [inline Content.Server/Prayer/PrayerSystem.cs:119]: Use AudioParams for volume and just overwrite the stream with stop + play.

File: Content.Shared/Prayer/PrayableComponent.cs [hunk 2]
```diff
@@ -30,6 +31,34 @@ public sealed partial class PrayableComponent : Component
...
+    [DataField("nextRing")]
+    [ViewVariables(VVAccess.ReadOnly)]
+    public TimeSpan NextRing;
+
+    /// <summary>
+    /// The cooldown on playing sounds for admins
+	/// </summary>
+    [DataField("ringPeriod")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan RingPeriod = TimeSpan.FromSeconds(10f);
```
Comment (metalgearsloth) [inline Content.Shared/Prayer/PrayableComponent.cs:60]: Datafields do not require strings anymore

File: Content.Shared/Prayer/PrayableComponent.cs [hunk 3]
```diff
@@ -30,6 +31,34 @@ public sealed partial class PrayableComponent : Component
     [ViewVariables(VVAccess.ReadWrite)]
     public string NotificationPrefix = "prayer-chat-notify-pray";
 
+    /// <summary>
+    /// The sound played for admins when a prayer is sent
+	/// </summary>
+    [DataField("soundForGod")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public SoundSpecifier SoundForGod = new SoundPathSpecifier("/Audio/Effects/plehnimda.ogg");
```
Comment (deltanedas) [inline Content.Shared/Prayer/PrayableComponent.cs:39]: all datafields are automatically rw now
and tag isnt needed

so just `[DataField]`

File: Content.Shared/Prayer/PrayableComponent.cs [hunk 4]
```diff
@@ -30,6 +31,34 @@ public sealed partial class PrayableComponent : Component
...
+    /// The volume of the sound played for admins when a prayer is sent
+	/// </summary>
+    [DataField("volumeForGod")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public int VolumeForGod = 0;
+	
+    /// <summary>
+    /// The last time a sound was played for admins
+	/// </summary>
+    [DataField("nextRing")]
```
Comment (deltanedas) [inline Content.Shared/Prayer/PrayableComponent.cs:51]: use time offset serializer here

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Shared/Prayer/PrayableComponent.cs (hunk 2 above)
Comment (metalgearsloth) [reply Content.Shared/Prayer/PrayableComponent.cs:60]: Also please remove the tabs and just use spaces

File: Content.Server/Prayer/PrayerSystem.cs (hunk 1 above)
Comment (IProduceWidgets) [reply Content.Server/Prayer/PrayerSystem.cs]: I could not figure this out. But in my testing it also did not work the way I was doing it, so I just decided to nuke the feature idea and pick .oggs that don't need adjusting.

**Comment by Emisse** [comment]
with how many noises appear already for admins in active rounds i think this is too much, these notifications really arent important enough to warrant this when most people notice them anyway in the chat bar

**Comment by IProduceWidgets** [comment]
I am not sure why the tests fail, but I don't think its my code. If someone has any idea, please let me know and Ill try it.

File: Content.Server/Prayer/PrayerSystem.cs (hunk 1 above)
Comment (TheShuEd) [reply Content.Server/Prayer/PrayerSystem.cs]: you can setup params in yml:

```yml
soundForGod:
  path: /Audio/Effects/voicemail.ogg
  params:
    pitch: ?
    variation: ?
    volume: ?
```

**Comment by AJCM-git** [comment]
@IProduceWidgets So from what i heard from admins they do want this but only if its toggleable, can you do that?

**Comment by retequizzle** [comment]
please include a toggle option or i will explode every person who sends prayers

on second thought

don't add a toggle i'd like an excuse to do that so we get less prayers

**Comment by IProduceWidgets** [comment]
Haha. I'll give it a go.
I haven't tried anything UI related yet, so it may take me a bit.

**Comment by metalgearsloth** [comment]
Majority opinion seems to be too annoying and it's just prayers anyway so.

**Pull request closed without merging**